hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/S3EventsHoodieIncrSource.java

```diff
@@ -110,6 +113,9 @@ public S3EventsHoodieIncrSource(
     if (!StringUtils.isNullOrEmpty(props.getString(Config.S3_KEY_PREFIX))) {
       filter = filter + " and s3.object.key like '" + props.getString(Config.S3_KEY_PREFIX) + "%'";
     }
+    if (!StringUtils.isNullOrEmpty(props.getString(Config.S3_IGNORE_KEY_PREFIX))) {
```

Comment:
apparently S3 incremental source does not have good tests. I have filed a tracking jira [here](https://issues.apache.org/jira/browse/HUDI-3089). Will take it as a follow up.

Reply:
will add the tests once https://github.com/apache/hudi/pull/4334 gets merged.